alchemist/alchemist-swingui/src/main/java/it/unibo/alchemist/boundary/gui/effects/DrawSmartcam.java

```diff
@@ -0,0 +1,84 @@
+package it.unibo.alchemist.boundary.gui.effects;
+
+import it.unibo.alchemist.model.implementations.actions.See;
+import it.unibo.alchemist.model.implementations.molecules.SimpleMolecule;
+import it.unibo.alchemist.model.implementations.positions.Euclidean2DPosition;
+import it.unibo.alchemist.model.interfaces.Environment;
+import it.unibo.alchemist.model.interfaces.Node;
+import it.unibo.alchemist.model.interfaces.Position;
+import it.unibo.alchemist.model.interfaces.environments.EuclideanPhysics2DEnvironment;
+import it.unibo.alchemist.model.interfaces.geometry.AwtShapeCompatible;
+import it.unibo.alchemist.model.interfaces.geometry.GeometricShape;
+
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.Shape;
+import java.awt.geom.AffineTransform;
+import java.awt.geom.Arc2D;
+
+/**
+ * Draws node's shapes and cameras' fields of view.
+ */
+public final class DrawSmartcam implements Effect {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public <T, P extends Position<P>> void apply(final Graphics2D g, final Node<T> node, final Environment<T, P> environment, final double zoom, final int x, final int y) {
+        if (!(environment instanceof EuclideanPhysics2DEnvironment)) {
```

Comment:
This is horrible: switch to positive logic, and log an error (once) in case of wrong environment.

Reply:
Done